src/core/geometry/geom3/retessellate.js

```diff
@@ -10,45 +9,43 @@ const fromPolygons = require('./fromPolygons')
   Polygons are split at each sweep line, and the fragments are joined horizontally and vertically into larger polygons
   (making sure that we will end up with convex polygons).
 */
-const retessellate = geometry => {
+const retessellate = (geometry) => {
+  if (!geometry.isCanonicalized) {
+    throw new Error('geometry must be canonical, call canonicalize first')
```

Comment:
nice !

Reply:
hmm or perhaps it should canonicalize by itself ? what do you think @z3dev

Reply:
I'll add a note to the new issue for the fuzzy factory. The rewrite should make this even easier, possibly even merge the logic into retessellate.

Reply:
Added comment to #139